bio/samtools/sort/wrapper.py

```diff
@@ -15,7 +15,12 @@
 extra = snakemake.params.get("extra", "")
 log = snakemake.log_fmt_shell(stdout=True, stderr=True)
 
-mem_per_thread_mb = int(get_mem(snakemake) / snakemake.threads)
+# `-m`: Approximately the maximum required memory per thread [...]
+# samtools sort can use more memory than specified (e.g. https://github.com/samtools/samtools/issues/831)
+mem_overhead_factor = snakemake.params.get("mem_overhead_factor", 0.1)
```

Comment:
The thread you link to suggests only using 75% of the available memory for the `-m` option, so maybe this would be the more conservative option:

```suggestion
mem_overhead_factor = snakemake.params.get("mem_overhead_factor", 0.25)
```

Or have you tested this and have found that `0.1` should usually be enough?